docs/06-javascript-interop.md

```diff
@@ -48,4 +48,57 @@ When the `OrderDetails` component polls for order status updates, an update set
 
 ![Real-time pizza map](https://user-images.githubusercontent.com/1874516/51807322-6018b880-227d-11e9-89e5-ef75f03466b9.gif)
 
+## Add a confirm prompt for deleting pizzas
+
+The JavaScript interop code for the `Map` component was provided for you. Next you'll add some JavaScript interop code of your own.
+
+It would be a shame if users accidentally deleted pizzas from their order (and ended up not buying them!). Let's add a confirm prompt when the user tries to delete a pizza. We'll show the confirm prompt using JavaScript interop.
+
+Add a static `JSRuntimeExtensions` class to the Client project with a `Confirm` extension method off of `IJSRuntime`. Implement the `Confirm` method to call the built-in JavaScript `confirm` function.
+
+```csharp
+    public static class JSRuntimeExtensions
+    {
+        public static Task<bool> Confirm(this IJSRuntime jsRuntime, string message)
+        {
+            return jsRuntime.InvokeAsync<bool>("confirm", message);
+        }
+    }
+```
+
+Inject the `IJSRuntime` service into the `Index` component so that it can be used there to make JavaScript interop calls.
+
+```
+@page "/"
+@inject HttpClient HttpClient
+@inject OrderState OrderState
+@inject IUriHelper UriHelper
+@inject IJSRuntime JS
+```
+
+Add an async `RemovePizza` method to the `Index` component that calls the `Confirm` method to verify if the user really wants to remove the pizza from the order.
+
+```csharp
+async Task RemovePizza(Pizza configuredPizza)
+{
+    if (await JS.Confirm($"Remove {configuredPizza.Special.Name} pizza from the order?"))
+    {
+        OrderState.RemoveConfiguredPizza(configuredPizza);
+    }
+}
+```
+
+Update the event handler for the `ConfiguredPizzaItems` to call the new `RemovePizza` method.
+
+```csharp
+@foreach (var configuredPizza in OrderState.Order.Pizzas)
+{
+    <ConfiguredPizzaItem Pizza="configuredPizza" OnRemoved="@(() => RemovePizza(configuredPizza))" />
```

Comment:
Would it be better to change `OnRemoved` to `OnRemovedAsync` so you can leave this line unchanged?

Reply:
I believe this link needs to change regardless, because adding the JS interop call to show a confirm dialog is a presentation related change that I believe needs to live in the component (not the OrderState). Do you agree?

However, it does seem like the OnRemove event on ConfiguredPizzaItem should now be a `Func<Task>` to support async. Is that correct?